### ISSUE TITLE
EFS (Elastic File System)

### ISSUE THREAD

**Issue by raphaelIl** [Raphael9292/study#1]
## TL;DR
- Network File System을 이용한 aws managed service

## 동작 방식
- https://docs.aws.amazon.com/ko_kr/efs/latest/ug/how-it-works.html#how-it-works-ec2

## 고려 사항
- efs 성능은 사이즈에 비례한다.
  - https://docs.aws.amazon.com/ko_kr/efs/latest/ug/performance.html#performance-overview
- efs에서 bursting mode는 버스트 크레딧에 영향을 받는다.
  - https://docs.aws.amazon.com/ko_kr/efs/latest/ug/performance.html#throughput-modes
- 그래서 efs 생성 후 덤프파일(1TiB)을 미리 생성하여 성능을 확보하는데 일반적이다.
  - https://docs.aws.amazon.com/ko_kr/efs/latest/ug/performance.html#efs-burst-credits

### self-hosted runner에서 빌드캐시로 사용할 무언가를 알아보다가...
- https://www.44bits.io/ko/post/amazon-efs-on-kubernetes-by-using-efs-provisioner
- https://hyperconnect.github.io/2021/12/21/github-actions-for-everyone-3.html
- https://fe-developers.kakaoent.com/2022/220414-docker-cache/
- https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows
- https://github.com/actions/setup-java#caching-packages-dependencies
- https://github.com/kubernetes-sigs/aws-efs-csi-driver
